docs/modeling-your-data/modeling-your-data-with-dbt/dbt-advanced-usage/dbt-utils-advanced-operation/index.md

```diff
@@ -7,3 +7,336 @@ sidebar_position: 500
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__user_identifiers: [{'schema': 'com_mycompany_user_identifier_1_1_0', 'field': 'user_id', 'prefix': 'ui_t'}, {'schema': 'com_mycompany_user_identifier_1_0_0', 'field': 'user_id', 'prefix': 'ui_o'}]
+    ...
+...
+```
+:::info
+Remember that the `prefix` key only needs to be set when running these models on Postgres or Redshift.
+:::
+
+### Combining atomic fields and custom contexts
+Combining atomic fields and custom contexts should hopefully then be straightforward if you're comfortable with what we described above. Let's say you want to combine the `logged_in_user_id` field from the `com_mycompany_logged_user_id_1_0_0` context together with the standard `domain_userid` field in the `events` table. To achieve that, you would include the following in your `dbt_profile.yml`:
+
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__user_identifiers: [{'schema': 'com_mycompany_logged_user_id_1_0_0', 'field': 'logged_in_user_id', 'prefix': 'lui'}, {'schema': 'atomic', 'field': 'domain_userid', 'prefix': 'e'}]
+    ...
+...
+```
+
+This would be parsed into the following SQL:
+
+```sql
+SELECT
+    ...
+    COALESCE(lui_logged_in_user_id, domain_userid, NULL) as user_identifier,
+...
+```
+# Introducing custom SQL logic to every event
+If there are certain SQL transformations you want to apply to events that are being processed by Snowplow's dbt packages you can leverage the `snowplow__custom_sql` variable to write out custom SQL that will be included in your `base_events_this_run` table, which can then be leveraged for any of your subsequent tables.
+
+The process for this looks slightly different for Redshift & Postgres relative to the other databases. If you use Redshift & Postgres, please skip ahead to the warehouse specific section to understand how that works.
+
+## Utilizing custom contexts or SDEs
+Suppose you have a custom context called `com_mycompany_click_1_0_0` which contains a `click_id` that you want to concat with Snowplow's `domain_sessionid`. You could either do this concatenation when creating your own data models, but if you want to surface this to all of your data models downstream from Snowplow's initial data processing in it's dbt packages, you can add that transformation by adding the following to your `dbt_project.yml`:
+
+```yml
+# dbt_project.yml
+...
+vars:
+    ...
+    snowplow__custom_sql: "CONCAT(com_mycompany_click_1_0_0.click_id, CONCAT('_', domain_sessionid)) as click_session_id"
+    ...
+...
+```
+
+If you'd like to add multiple lines of SQL, you can do that as well by making this string a multi-line string. Any SQL included in the `snowplow__custom_sql` will be found in your `snowplow_base_events_this_run` table, and will be referenced at the end of the select statement, so there's no need to add a trailing comma. Any of the newly created fields can also be passed through to other tables created automatically by Snowplow's dbt packages.
+
```

Comment:
That feels a bit less intuitive to me, and multi-line is really just to make it pretty. But to me it makes the most intuitive sense to just write the SQL that you want to appear, and then it appears, rather than split it into an entry in a list per line